packages/w3up-client/src/client.js

```diff
@@ -233,12 +235,50 @@ export class Client extends Base {
 
   /**
    * Create a new space with a given name.
+   * If an account is provided in the options argument, then it creates a delegated recovery account
+   * by provisioning the space and then delegating access to the recovery account.
+   *
+   * @typedef {object} CreateOptions
+   * @property {Account.Account} [account]
    *
    * @param {string} name
+   * @param {CreateOptions} options
+   * @returns {Promise<import("./space.js").OwnedSpace>} The created space owned by the agent.
    */
-  async createSpace(name) {
-    return await this._agent.createSpace(name)
+  async createSpace(name, options = {}) {
+    const space = await this._agent.createSpace(name)
+
+    const account = options.account
+    if (account) {
+      // Provision the account with the space
+      const provisionResult = await account.provision(space.did())
+      if (provisionResult.error) {
+        throw new Error(
+          `⚠️ Failed to provision account: ${provisionResult.error.name}:${provisionResult.error.message}`
+        )
+      }
+
+      // Save the space to authorize the client to use the space
+      await space.save()
```

Comment:
Not sure about this - if I don't pass an `account` the space is not saved to my store...should it always be saved?

I think it's ok to only save if an account is passed but it needs to be mentioned in the jsdoc.

If you switch to always saving the space perhaps add a `noSave` option to retain the ability for ephemeral spaces to be created?

Reply:
I believe it makes sense to save only if you provide the `account`, mainly because if you provide the account we need to get the authorizations done via `.save` call to be able to delegate to the recovery. Then if you want to create an ephemeral space you don't need the account because there is no reason to create a recovery. I will update the `jsdoc` to take that into account 😃